yt/frontends/gadget/data_structures.py

```diff
@@ -546,6 +546,7 @@ def _is_valid(self, *args, **kwargs):
             fh = h5py.File(args[0], mode='r')
             valid = all(ng in fh["/"] for ng in need_groups) and \
                 not any(vg in fh["/"] for vg in veto_groups)
+            valid = fh["Header"].attrs["Code"] != b"SWIFT"
```

Comment:
I would prefer it if you would convert the stuff coming out of hdf5 to utf-8; the specific formatting might change in the future.

Reply:
so what do you mean here? h5py loads this as a numpy._bytes which is in the utf-8 format. What would you prefer here?

Reply:
`valid = fh["Header"].attrs["Code"].decode("utf-8") != "SWIFT"`